qml/pages/CameraUI.qml

```diff
@@ -207,37 +175,229 @@ Page {
+            rotation: page.controlsRotation
+            transformOrigin: {
+                if (camera._orientation == 1 || camera._orientation >= 5) return Item.Right
+                else if (camera._orientation == 2) return Item.Left
+                else return Item.Center
+            }
+
+            Label {
+                property bool forceUpdate: false
+                id: lblResolution
+                color: Theme.lightPrimaryColor
+                text: (forceUpdate || !forceUpdate) ? settings.sizeToStr(settings.resolution(settings.global.captureMode)) : ""
+            }
+
+            Label {
+                id: lblRecordTime
+                visible: settings.global.captureMode == "video"
+                color: Theme.lightPrimaryColor
+                //text: Qt.formatDateTime(new Date(camera.videoRecorder.duration), "hh:mm:ss") //Doest work as return 01:00:00 for 0
+                text: msToTime(camera.videoRecorder.duration)
+            }
+        }
+
+        SettingsOverlay {
+            id: settingsOverlay
+            iconRotation: page.controlsRotation
+        }
+
+        RoundButton {
+            id: btnGallery
+
+            visible: galleryModel.count > 0
+            enabled: visible
+
+            anchors.top: btnCameraSwitch.bottom
+            anchors.bottomMargin: Theme.paddingMedium
+            anchors.right: parent.right
+            anchors.rightMargin: Theme.paddingMedium
+            icon.rotation: page.controlsRotation
+
+            size: Theme.itemSizeSmall
+
+            image: "image://theme/icon-m-image"
+
+            onClicked: {
+                camera.stop();
+                pageStack.push(Qt.resolvedUrl("GalleryUI.qml"), { "fileList": galleryModel });
+            }
+        }
+
+        RoundButton {
+            id: btnCameraSwitch
+            icon.source: "image://theme/icon-camera-switch"
+            visible: settings.global.cameraCount > 1
+            icon.rotation: page.controlsRotation
+            anchors {
+                top: parent.top
+                topMargin: Theme.paddingMedium
+                right: parent.right
+                rightMargin: Theme.paddingMedium
+            }
+            onClicked: {
+                _cameraReload = true;
+                console.log("Setting temp resolution");
+                _setTempResolution = true;
+                camera.imageCapture.setResolution(settings.strToSize("320x240"));
+                camera.stop();
+                camera.deviceId = settings.global.cameraId == "primary" ? "secondary" : "primary";
+                _parametersLoaded = false;
+                tmrDelayedStart.start();
+            }
+        }
+
+        IconSwitch {
+            id: btnModeSwitch
+            anchors.bottom: parent.bottom
+            anchors.bottomMargin: Theme.paddingMedium
+            anchors.right: parent.right
+            anchors.rightMargin: (rotation == 90 || rotation == -90) ? Theme.paddingLarge * 2 : Theme.paddingMedium
```

Comment:
Change -90 to 270 to prevent lower icon from being partially outside screen area in portrait mode.